### ISSUE TITLE
[BUG] IQ scan not finding dependencies when using gradle (or gradlew)

### ISSUE THREAD

**Issue by rpanman-sonatype** [sonatype-nexus-community/vscode-iq-plugin#224]
**Describe the bug**
When running a scan on a gradle project with dependencies in the gradle.build file, no dependencies are found and the VSCode IQ plugin returns no results.

**To Reproduce**
1. Clone a project that uses gradle onto your local machine:
```git clone https://github.com/sonatype-nexus-community/nexusiq-successmetrics.git```
2. Open the project in VSCode
3. Configure a .sonatype-config file to define the application name
3. Kick off a VSCode IQ plugin scan the project - information is sent to the IQ server and received back
4. VSCode IQ plugin will show no dependencies (and hence no analysis in the scan results) despite there being dependencies in build.gradle.

**Expected behavior**
VSCode IQ plugin finds the dependencies and presents those to the IQ server for analysis, the results are then displayed correctly in the "Sonatype Scan Results" section.

**Screenshots**
Empty scan results:
![image](https://user-images.githubusercontent.com/95854467/148558601-253549d8-8820-4e27-a5ec-702f71b500bf.png)

**Desktop (please complete the following information):**
 - OS: Mac OS X
 - NodeJS Version: unknown
 - VS Code Version: 1.63.2
 - Version of Extension v1.2.0
 - Gradle version: 7.3.3 (the problem also seems to happen with gradlew which is version 7.2 in the github project above)

**Additional context**
Detailed logs shows no dependencies found:
```
{"startTime":"2022-01-07T13:25:56.246Z","categoryName":"sonatypeIQ","data":["Full gradle command constructed","gradle dependencies --configuration runtimeClasspath"],"level":{
"level":20000,"levelStr":"INFO","colour":"green"},"context":{},"pid":9104},
{"startTime":"2022-01-07T13:25:56.246Z","categoryName":"sonatypeIQ","data":["Attempting to run gradle command","gradle dependencies --configuration runtimeClasspath"],"level":
{"level":10000,"levelStr":"DEBUG","colour":"cyan"},"context":{},"pid":9104},
{"startTime":"2022-01-07T13:25:58.440Z","categoryName":"sonatypeIQ","data":["Gradle command has run","gradle dependencies --configuration runtimeClasspath"],"level":{"level":1
0000,"levelStr":"DEBUG","colour":"cyan"},"context":{},"pid":9104},
{"startTime":"2022-01-07T13:25:58.441Z","categoryName":"sonatypeIQ","data":["Obtained 0 Dependencies from Muncher GradleDependencies",[]],"level":{"level":5000,"levelStr":"TRACE","colour":"blue"},"context":{},"pid":9104},
```

The gradle command above does return dependencies:
```
➜  nexusiq-successmetrics git:(playing-remove-current-time) ✗ gradle dependencies --configuration runtimeClasspath

> Task :dependencies

------------------------------------------------------------
Root project 'successmetrics'
------------------------------------------------------------

runtimeClasspath - Runtime classpath of source set 'main'.
+--- org.springframework.boot:spring-boot-devtools -> 2.5.6
|    +--- org.springframework.boot:spring-boot:2.5.6
|    |    +--- org.springframework:spring-core:5.3.12
|    |    |    \--- org.springframework:spring-jcl:5.3.12
|    |    \--- org.springframework:spring-context:5.3.12
|    |         +--- org.springframework:spring-aop:5.3.12
|    |         |    +--- org.springframework:spring-beans:5.3.12
|    |         |    |    \--- org.springframework:spring-core:5.3.12 (*)
|    |         |    \--- org.springframework:spring-core:5.3.12 (*)
|    |         +--- org.springframework:spring-beans:5.3.12 (*)
|    |         +--- org.springframework:spring-core:5.3.12 (*)
|    |         \--- org.springframework:spring-expression:5.3.12
|    |              \--- org.springframework:spring-core:5.3.12 (*)
|    \--- org.springframework.boot:spring-boot-autoconfigure:2.5.6
|         \--- org.springframework.boot:spring-boot:2.5.6 (*)
...
```

cc @bhamail / @DarthHater

**Comment by madpah** [comment]
Thanks for the report @rpanman-sonatype.

As discussed on Slack, this seems to be related to a `gradle` version issue.

Currently the plugin favours a global `gradle` command over `./gradlew` which may be in your project.

@DarthHater - would you have any thoughts / objection on us switching to favouring `./gradlew` over `gradle`?

FYI: @ctownshend / @hboutemy

**Comment by rpanman-sonatype** [comment]
Strangely if I uninstall gradle, I can see that the plugin picks up gradle**w** however the result is the same, nothing in the results pane.

**Comment by madpah** [comment]
That is strange @rpanman-sonatype.

Going to solicit some advice from @DarthHater before we make any changes.

**Comment by rpanman-sonatype** [comment]
I started some more in depth troubleshooting. It seems that parseGradleDependencyTree is called with includeDev set to true - parseGradleDependencyTree then looks for 'testRuntimeClasspath' as the start string for determining the list of depenencies. Since this string is not present in the gradle output then no dependencies are found. Next step is to work out why includeDev is set to true.

**Comment by madpah** [comment]
`includeDev` is a feature that you can toggle in the plugin:

![Screenshot 2022-01-07 at 16 32 40](https://user-images.githubusercontent.com/10280392/148575350-b9671fc4-e94f-43df-925f-a2ef6555e16e.png)

Do you want to try disabling that @rpanman-sonatype ?

**Comment by rpanman-sonatype** [comment]
Yes, that fixes it. So now I'm wondering why the gradle dependencies command is `gradle dependencies --configuration runtimeClasspath`- shouldn't is use `testRuntimeClasspath` rather than `runtimeClasspath`?
